src/super_gradients/training/datasets/obb/dota.py

```diff
@@ -0,0 +1,340 @@
+import multiprocessing
+import random
+import cv2
+import numpy as np
+
+from functools import partial
+from pathlib import Path
+from typing import Tuple, Iterable
+
+from super_gradients.module_interfaces import HasPreprocessingParams
+from super_gradients.training.datasets.data_formats.obb.cxcywhr import poly_to_cxcywhr
+from tqdm import tqdm
+
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.object_names import Processings
+from super_gradients.common.registry import register_dataset
+from super_gradients.dataset_interfaces import HasClassesInformation
+from super_gradients.training.transforms import OBBDetectionCompose
+from super_gradients.training.transforms.obb import OBBSample
+from torch.utils.data import Dataset
+from super_gradients.common.factories.transforms_factory import TransformsFactory
+
+__all__ = ["DOTAOBBDataset"]
+
+
+@register_dataset()
+class DOTAOBBDataset(Dataset, HasPreprocessingParams, HasClassesInformation):
```

Comment:
Does inheriting from detectiondataset not make sense ?
In case it doe not, I believe we should introduce a base class for OBB at least.

Reply:
Fixed in 3427148c1f8114a7252e2f0597cd746de00d2bb9